rbac/management/tenant_service/v2.py

```diff
@@ -103,6 +104,94 @@ def update_user(
+            )
+            workspaces.append(
+                Workspace(
+                    uuid=default_workspace_uuid, tenant=tenant, type=Workspace.Types.DEFAULT, name="Default Workspace"
+                )
+            )
+            # Existing tenant might have mapping or not
+            if not hasattr(tenant, "tenantmapping"):
+                mappings_to_create.append(TenantMapping(tenant=tenant))
+            else:
+                tenant_mappings[tenant.id] = tenant.tenantmapping
+            tenant_id = f"{self._user_domain}/{tenant.org_id}"
+            relationships = [
+                create_relationship(
+                    ("rbac", "workspace"),
+                    str(default_workspace_uuid),
+                    ("rbac", "workspace"),
+                    str(root_workspace_uuid),
+                    "parent",
+                ),
+                create_relationship(
+                    ("rbac", "workspace"), str(root_workspace_uuid), ("rbac", "tenant"), tenant_id, "parent"
+                ),
+            ]
+
+            # Include platform for tenant
+            relationships.append(
+                create_relationship(("rbac", "tenant"), tenant_id, ("rbac", "platform"), settings.ENV_NAME, "platform")
+            )
+
+        # Ignore those already created
+        Workspace.objects.bulk_create(workspaces, ignore_conflicts=True)
+
+        mappings = TenantMapping.objects.bulk_create(mappings_to_create, ignore_conflicts=True)
+        tenant_mappings.update({mapping.tenant_id: mapping for mapping in mappings})
+        bootstrapped_tenants = []
+
+        # Get existing custom groups
+        default_group_uuids = [mapping.default_group_uuid for mapping in tenant_mappings.values()]
+        existing_custom_groups = set(Group.objects.filter(uuid__in=default_group_uuids).values_list("uuid", flat=True))
+        for tenant, default_workspace_uuid in zip(tenants, default_workspace_uuids):
+            mapping = tenant_mappings[tenant.id]
+            relationships.extend(
+                self._bootstrap_default_access(mapping, str(default_workspace_uuid), existing_custom_groups)
+            )
+            bootstrapped_tenants.append(BootstrappedTenant(tenant, mapping))
+        self._replicator.replicate(
+            ReplicationEvent(
+                event_type=ReplicationEventType.BOOTSTRAP_TENANT,
+                info={"CJI": "bulk_bootstrap"},
+                partition_key="rbactodo",
+                add=relationships,
+            )
+        )
+        return bootstrapped_tenants
+
+    def _get_or_bootstrap_tenants(self, org_ids: list) -> list[BootstrappedTenant]:
+        """Bootstrap list of tenants, used by update_users."""
+        # Fetch existing tenants
+        existing_tenants = {
+            tenant.org_id: tenant
+            for tenant in Tenant.objects.filter(org_id__in=org_ids).select_related("tenantmapping")
```

Comment:
Do you think we should set `related_name` on the `tenant` field of `TenantMapping` model to `tenant_mapping` to make this fit in with other related names?

Reply:
Yea, setting it looks better. I will add a migration for it

Reply:
Oh, this changes the column name? Bummer but I guess okay.

Reply:
Yea, the column name. Right, just a small migration👍